### ISSUE TITLE
atlas-cluster Helm Chart Should Return Only After Cluster Creation

### ISSUE THREAD

**Issue by alex-tw-lam** [mongodb/mongodb-atlas-kubernetes#235]
Currently, the `mongodb/atlas-cluster` helm chart will return once the `atlasclusters.atlas.mongodb.com` resource is deployed: 

```
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
atlas-cluster   default         1               2021-05-20 12:28:40.202202 +0800 HKT    deployed        atlas-cluster-0.1.5             0.5.0
atlas-operator  atlas-operator  1               2021-05-20 12:28:19.402321 +0800 HKT    deployed        mongodb-atlas-operator-0.2.1    0.5.0

```

However, the desired cluster and database users are still in `CREATING` state: 

```
{
  "conditions": [
    {
      "lastTransitionTime": "2021-05-20T04:28:41Z",
      "status": "False",
      "type": "Ready"
    },
    {
      "lastTransitionTime": "2021-05-20T04:28:43Z",
      "message": "cluster is provisioning",
      "reason": "ClusterCreating",
      "status": "False",
      "type": "ClusterReady"
    }
  ],
  "observedGeneration": 1,
  "stateName": "CREATING"
}
```

Ideally, the helm chart should have a status of 'pending-install' when the cluster is still being created.

**Comment by alex-tw-lam** [comment]
Likewise, a `helm uninstall` should wait for `DELETING` of the `atlas-cluster`

**Comment by theburi** [comment]
helm status shows the status of a HELM chart not a service under its management. you could use --wait to wait for underlying resources to get to a ready state.

**Comment by alex-tw-lam** [comment]
Already waited on the `helm install`. Here's the result @theburi : 

```
Status:
  Conditions:
    Last Transition Time:  2021-05-20T12:35:38Z
    Status:                False
    Type:                  Ready
    Last Transition Time:  2021-05-20T12:35:39Z
    Message:               cluster is provisioning
    Reason:                ClusterCreating
    Status:                False
    Type:                  ClusterReady
  Observed Generation:     1
  State Name:              CREATING
```

**Comment by jasonmimick** [comment]
Hi @alex-tw-lam -
Our atlas-cluster chart still exhibits this behavior. As with other database services, these are longer running operations for provisioning. Could you let us know if support _specifically_ `helm --wait` is a requirement for your project?

**Comment by jasonmimick** [comment]
Since this issue is stale - closing.
Feel free open new issue if this is still a concern.
